### PR TITLE
chore: hide `__safeContent__` field COMPASS-5780

### PR DESCRIPTION
**Pull request by addaleax**
Grey out the field in the CRUD view, mark it non-editable, and
deselect it by default when exporting from a collection.

Screenshot from 2022-05-03 16-15-37
Screenshot from 2022-05-03 16-15-49
Screenshot from 2022-05-03 16-27-39

### Checklist
- [ ] New tests and/or benchmarks are included
- [ ] Documentation is changed or added

## Motivation and Context

- [ ] Bugfix
- [ ] New feature
- [ ] Dependency update
- [x] Misc

## Types of changes

- [ ] *Backport Needed*
- [x] Patch (non-breaking change which fixes an issue)
- [ ] Minor (non-breaking change which adds functionality)
- [ ] Major (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: packages/compass-import-export/src/modules/load-fields.js [hunk 1]
```diff
@@ -54,5 +54,8 @@ export function getSelectableFields(fields, { maxDepth } = {}) {
   const selectableFields = Object.keys(fields).map((field) =>
     truncateFieldToDepth(field, maxDepth)
   );
-  return Object.fromEntries(selectableFields.map((field) => [field, ENABLED]));
+  return Object.fromEntries(selectableFields.map((field) => {
+    const enabled = !/^__safeContent__\b/.test(field);
```
Comment (Anemy) [inline packages/compass-import-export/src/modules/load-fields.js:58]: I'm thinking this is worth a comment since the `__safeContent__` field might be a bit unexpected. Would it help to break this into a helper so we can test it independently?
The regex looks good, I think it makes me sleep good when they're tested though.

I think you mentioned that this internal field is useful in mongo dump/restore. Is that relevant to importing/exporting?

```suggestion
    // The `__safeContent__` field is used internally in MongoDB and can cause
    // issues when it is duplicated or changed by users.
    // We default to not export it unless the user explicitly enables it.
    const enabled = !/^__safeContent__\b/.test(field);
```

**Review by Anemy — COMMENTED** [review]
Should we hide this field on the `schema` page as well?

File: packages/compass-import-export/src/modules/load-fields.js (hunk 1 above)
Comment (mcasimir) [reply packages/compass-import-export/src/modules/load-fields.js:58]: I'm actually thinking if the idea of having that as a projection in data service is not better here, we can figure ways to share the logic with DE too, maybe we just need to keep that in mind?

Comment (addaleax) [reply packages/compass-import-export/src/modules/load-fields.js]: @Anemy Just fyi, I restructured this a bit so that the logic for determining this field handling is more bundled in hadron-document – I think that makes sense here, I also really hope that we don’t ever have a field like this introduced by the server, but who knows :woman_shrugging: 

@mcasimir I mentioned this on Slack already, but yeah, I think it’s too confusing to have fields that are displayed only under some circumstances, and filtering in the data service is also complicated because there are situations, like the export one, in which a user might have a good reason to want to include the field depending on the situation

**Review by Anemy — APPROVED** [review]
